tests/rgen/Microsoft.Macios.Generator.Tests/Classes/Data/Trampolines/ExpectedAVFoundationTrampolinesTrampolines.cs

```
@@ -0,0 +1,1369 @@
+				return default (global::ObjCRuntime.BlockLiteral);
+			return CreateBlock (callback);
+		}
+
+		[BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
+		internal static unsafe global::ObjCRuntime.BlockLiteral CreateBlock (global::AVFoundation.AVAssetPlaybackAssistantLoadPlaybackConfigurationOptionsHandler callback)
+		{
+			delegate* unmanaged<global::System.IntPtr, global::ObjCRuntime.NativeHandle, void> trampoline = &Invoke;
+			return new global::ObjCRuntime.BlockLiteral (trampoline, callback, typeof (SDAVAssetPlaybackAssistantLoadPlaybackConfigurationOptionsHandler), nameof (Invoke));
+		}
+	}
+	// TODO: generate trampoline for AVFoundation.AVAssetPlaybackAssistantLoadPlaybackConfigurationOptionsHandler
+
+	[UnmanagedFunctionPointerAttribute (CallingConvention.Cdecl)]
+	[UserDelegateType (typeof (global::AVFoundation.AVAudioConverterInputHandler))]
+	unsafe internal delegate global::ObjCRuntime.NativeHandle DAVAudioConverterInputHandler (global::System.IntPtr block_ptr, uint inNumberOfPackets, global::System.IntPtr* outStatus);
+
+	/// <summary>This class bridges native block invocations that call into C#</summary>
+	static internal class SDAVAudioConverterInputHandler
+	{
+		[Preserve (Conditional = true)]
+		[UnmanagedCallersOnly]
+		[UserDelegateType (typeof (global::AVFoundation.AVAudioConverterInputHandler))]
+		internal static unsafe global::ObjCRuntime.NativeHandle Invoke (global::System.IntPtr block_ptr, uint inNumberOfPackets, global::System.IntPtr* outStatus)
+		{
+			*outStatus = default;
+			var del = global::ObjCRuntime.BlockLiteral.GetTarget<global::AVFoundation.AVAudioConverterInputHandler> (block_ptr);
+			if (del is not null)
+			{
+				var ret = del (inNumberOfPackets, out global::System.Runtime.CompilerServices.Unsafe.AsRef<global::AVFoundation.AVAudioConverterInputStatus> (outStatus));
+				return global::ObjCRuntime.Runtime.RetainAndAutoreleaseNSObject (ret);
+			}
+			return default;
+		}
```

bgen code:
```csharp
[Preserve (Conditional = true)]
[UnmanagedCallersOnly]
[UserDelegateType (typeof (global::AVFoundation.AVAudioConverterInputHandler))]
internal static unsafe NativeHandle Invoke (IntPtr block, uint inNumberOfPackets, global::AVFoundation.AVAudioConverterInputStatus* outStatus) {
	var del = BlockLiteral.GetTarget<global::AVFoundation.AVAudioConverterInputHandler> (block);
	var retval = del (inNumberOfPackets, out global::System.Runtime.CompilerServices.Unsafe.AsRef<global::AVFoundation.AVAudioConverterInputStatus> (outStatus));
	return Runtime.RetainAndAutoreleaseNSObject (retval);
}
```
There are a few diffs here:
1. rgen inits the outStatus to the default.
2. rgen checks if the del is null, it it is, it returns the default.


AFAIk we need to always check for del to be null, since BlockLiteral.GetTarget returns a nullable value, which means that we always need to init the out variable to its default and we need to make sure we return the default of NativeHandle which AFAIK is a null ptr.

---

If `BlockLiteral.GetTarget` returns null, something went very wrong, so IMHO it's better to throw an exception in that case.